gui/src/app/pages/HomePage/HomePage.tsx

```diff
@@ -93,6 +95,20 @@ const HomePage: FunctionComponent<Props> = ({ width, height }) => {
         window.location.reload()
     }, [])
 
+    const [sampler, setSampler] = useState<StanSampler | undefined>(undefined);
+
+    useEffect(() => {
+        if (!compileMainJsUrl) {
+            setSampler(undefined);
+            return;
+        }
+        const s = new StanSampler(compileMainJsUrl);
+        setSampler(s);
+        return () => {
+            s.cancel();
```

Comment:
I don't think this will work, since `s.cancel()` will end up re-calling initialize?

Have you verified there aren't dangling threads after this?

Edit: there do appear to be in the firefox debugger

Reply:
You're right. I changed it to terminate. Now the class has two methods: cancel() and terminate(). The latter does not restart the worker thread.

Reply:
I am still seeing the number of threads grow after each compilation

Reply:
I made some adjustments to StanSampler and now this should be fixed. The problem was that an additional call to cancel() was being made which reinitialized the worker. Now cancel() only has any effect if the status was 'sampling'.

Reply:
I can confirm there is no extra threads now, but I am getting the "Nothing to cancel" warning, so I think there is still a call being made that shouldn't be there, it's just been turned into a no-op now?

Reply:
Thanks, I have now removed that extra no-op call.